xrally_kubernetes/service.py

```diff
@@ -1117,6 +1119,157 @@ def delete_statefulset(self, name, namespace, status_wait=True):
                                    resource_type="StatefulSet",
                                    namespace=namespace)
 
+    @atomic.action_timer("kubernetes.list_node")
+    def list_node(self):
+        return self.v1_client.list_node()
+
+    def list_filtered_nodes(self, node_labels=None):
+        """Return list of optionally filtered nodes names.
+
+        :param node_labels: map, each key is a label name with some value
+        """
+        node_meta = [node.metadata for node in self.list_node().items]
+        if node_labels is None:
+            return [meta.name for meta in node_meta]
+
+        node_names = []
+        for meta in node_meta:
+            for k, v in meta.labels.items():
+                if k in node_labels and node_labels[k] == v:
+                    node_names.append(meta.name)
+        return node_names
+
+    @atomic.action_timer("kubernetes.get_daemonset")
+    def get_daemonset(self, name, namespace, **kwargs):
+        return self.v1beta1_ext.read_namespaced_daemon_set(
+            name,
+            namespace=namespace
+        )
+
+    @atomic.action_timer("kubernetes.create_daemonset")
+    def create_daemonset(self, namespace, image, command=None,
+                         node_labels=None, status_wait=True):
+        """Create daemon set and optionally wait for status.
+
+        :param namespace: daemon set namespace
+        :param image: daemon set template image
+        :param command: daemon set template command
+        :param node_labels: map, each key is a label name with some value
+        :param status_wait: wait for status if True
+        :return: name and app
+        """
+        name = self.generate_random_name()
+        app = self.generate_random_name()
+
+        container_spec = {
+            "name": name,
+            "image": image
+        }
+        if command is not None and isinstance(command, (list, tuple)):
```

Comment:
I prefer to raise ValueError in case of `not isinstance(command, (list, tuple))`

Reply:
Maybe better in separate cleanup patch?